File: bestia-models/src/test/java/net/bestia/util/ObjectSerializerTest.java
package net.bestia.util;

import net.bestia.model.util.ObjectSerializer;
import org.junit.Assert;
import org.junit.Before;
import org.junit.Test;


public class ObjectSerializerTest {
	
	private ObjectSerializer<String> ser;
	
	@Before
	public void setup() {
		
		ser = new ObjectSerializer<>();
	}

	@Test(expected = NullPointerException.class)
	public void deserialize_null_throws() {
		ser.deserialize(null);
	}


	@Test(expected = NullPointerException.class)
	public void serialize_null_throws() {
		ser.serialize(null);
	}

	@Test
	public void serializeAndDeserialize_validStream_object() {
		final String dataObj = "Hello World";
		byte[] data = ser.serialize(dataObj);
		Assert.assertNotNull(data);
		final String newDataObj = ser.deserialize(data);
		Assert.assertEquals(dataObj, newDataObj);
	}

}

File: bestia-models/src/test/kotlin/net/bestia/util/ObjectSerializerTest.kt
package net.bestia.util

import net.bestia.model.util.ObjectSerializer
import java.util.Arrays

import org.junit.Assert
import org.junit.Test

class ObjectSerializerTest {
  @Test
  fun deserialize_invalidData_null() {
    val invalidData = ByteArray(10)
    Arrays.fill(invalidData, 0x12.toByte())

    val test: String? = ObjectSerializer.deserialize(invalidData)
    Assert.assertNull(test)
  }

  @Test
  fun serializeAndDeserialize_validObject_ok() {
    val test = "Hello World"
    val data = ObjectSerializer.serialize(test)

    Assert.assertNotNull(data)
    Assert.assertTrue(data!!.isNotEmpty())

    val test2 = ObjectSerializer.deserialize<String>(data)

    Assert.assertEquals(test, test2)
  }
}
